boards/nrf52840-dongle/Makefile.dep

```diff
@@ -0,0 +1,25 @@
+ifneq (,$(filter saul_default,$(USEMODULE)))
+  USEMODULE += saul_gpio
+endif
+
+# It'd be nice to pull this in on any stdin dependency and not just shell, but
+# that's the common one, and by the time shell has pulled in stdin, the default
+# stdio_uart has already been set.
+#
+# Maybe have a shell_default like netdev_default?
+ifneq (,$(filter shell,$(USEMODULE)))
```

Comment:
I'm not sure I understand, even with the comment, why this is dependent on the shell.

Reply:
I *did* seem to have some mixup between stdin and stdio, revisiting that in more detail. (When I wrote this I was unaware that stdin is extra functionality atop of sdio)

The idea was: If an application depends on stdio (whether directly or via shell, but I though I couldn't catch the "directly" case), it will pull in a stdio implementation. A board should be able to provide a default stdio, but may want (kind of like netdev_default does), but an application may still want to use stdio_rtt on a USB-supporting board explicitly.

But as said, I'll go through this again with better awareness of stdio vs stdin.